### PR TITLE
[NEEDS #invenio-vocabularies/pull/199 #invenio-rdm-records/pull/979]recources: add ui serializer

### PR DESCRIPTION
**Pull request by jrcastro2**
* closes https://github.com/inveniosoftware/invenio-app-rdm/issues/1518
* needs https://github.com/inveniosoftware/invenio-rdm-records/pull/979

### CODE REVIEW

File: invenio_communities/assets/semantic-ui/js/invenio_communities/settings/profile/index.js [hunk 1]
```diff
@@ -423,6 +423,9 @@ class CommunityProfileForm extends Component {
     // Clean values
     submittedCommunity = removeEmptyValues(submittedCommunity);
 
+    // Remove ui object
+    delete submittedCommunity.ui;
```
Comment (kpsherva) [inline invenio_communities/assets/semantic-ui/js/invenio_communities/settings/profile/index.js:427]: isn't it better to make the `.ui` key dump_only and let the backend handle it? it is going to be quite problematic to always remember to remove it in the frontend

File: invenio_communities/views/decorators.py [hunk 2]
```diff
@@ -11,18 +11,33 @@
...
+        @wraps(f)
+        def view(**kwargs):
+            pid_value = kwargs["pid_value"]
+            community = current_communities.service.read(
+                id_=pid_value, identity=g.identity
+            )
+            kwargs["community"] = community
+            if serialize:
+                community_ui = CustomJSONSerializer(
+                    object_key='ui',
```
Comment (kpsherva) [inline invenio_communities/views/decorators.py:34]: I find it a bit complex to specify the "ui" key on each usage, I had an impression we were trying to "standarize" it across the code base. How about having a new class inherited from `MarshmallowSerializer` which assigns these 4 arguments as class attributes? 
Is it possible to have it registered with the community record in the config? ping @ntarocco what do you think?

Comment (kpsherva) [reply invenio_communities/views/decorators.py:34]: after re-reading the PR, I see the response serializer is already registered in the resource config, but using it here in the decorator actually prevents the config from being used, I think it would be better to use the response serializer rather than implementing it here in the decorator

File: invenio_communities/views/decorators.py [hunk 3]
```diff
@@ -11,18 +11,33 @@
...
 from flask import g
+from flask_resources import CustomJSONSerializer, JSONSerializer, UIListSchema
 
+from invenio_communities.communities.resources.UISchema import \
+    UICommunitySchema
 from invenio_communities.proxies import current_communities
 
 
-def pass_community(f):
+def pass_community(serialize=False):
```
Comment (ntarocco) [inline invenio_communities/views/decorators.py:21]: I would avoid the default value `False`: this is because when using the decorator, if you don't pass the param (because your want it `False`), Flask will not be happy and will throw an error. It took me quite some time last time to debug it.
TL;DR: when a decorator as params, it must be always be called with the params.

File: invenio_communities/views/decorators.py (hunk 2 above)
Comment (ntarocco) [reply invenio_communities/views/decorators.py:34]: The one configured in the `config` is for REST APIs, I am not sure that you can easily reuse it here.
In any case, maybe we can indeed create a `CommunitySerializer` or similar, instantiating (or inheriting if not possible to compose) `MarshmallowSerializer` with the params here.

File: invenio_communities/views/decorators.py (hunk 3 above)
Comment (jrcastro2) [reply invenio_communities/views/decorators.py:21]: Ok, thanks a lot for sharing!

File: invenio_communities/assets/semantic-ui/js/invenio_communities/settings/profile/index.js (hunk 1 above)
Comment (jrcastro2) [reply invenio_communities/assets/semantic-ui/js/invenio_communities/settings/profile/index.js:427]: I am not 100% sure about this, I understand the point that it can be a bit more problematic but I think it should be the client responsible for sending some valid data and not the backend adapting to the client needs, note that the schema used to fetch the data is different than the one used to post the data, this is due to localization as we need to translate the vocabularies. That said we could easily add to the community Marshamallow schema `unknown = EXCLUDE` but this implies that when using the API we will not get an error if we add non-existing fields, I am fine with this, if this works for you I could add this change but want to double check with you first.

Please note that for the records we do also remove the `ui` object before sending the data  (see code [here](https://github.com/inveniosoftware/react-invenio-deposit/blob/master/src/lib/DepositRecordSerializer.js#L362-L370)). If we want to change this behaviour I think that we should do it for both. 

I am also available to discuss IRL if it's easier.

Comment (ntarocco) [reply invenio_communities/assets/semantic-ui/js/invenio_communities/settings/profile/index.js:427]: @slint @alejandromumo any idea why the `unknown = EXCLUDE` was not added? Simply forgotten?

Comment (slint) [reply invenio_communities/assets/semantic-ui/js/invenio_communities/settings/profile/index.js:427]: Probably yes, it was forgotten... This was one of the very first forms we did in React, so the common services/resources/schemas practices we have now were not that common.

File: invenio_communities/communities/resources/serializer.py [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+# under the terms of the MIT License; see LICENSE file for more details.
+
+"""Invenio Communities Resource Serializers."""
+
+from flask_resources import MarshmallowSerializer, UIListSchema, JSONSerializer
+
+from invenio_communities.communities.resources.UISchema import UICommunitySchema
+
+
+class UICommunityJSONSerializer(MarshmallowSerializer):
```
Comment (zzacharo) [inline invenio_communities/communities/resources/serializer.py:16]: nit: I would rename it to `CommunityUIJSONSerializer`. Not blocking :)

File: invenio_communities/views/decorators.py [hunk 5]
```diff
@@ -12,17 +12,31 @@
...
-
-    return view
+    def decorator(f):
+        @wraps(f)
+        def view(**kwargs):
+            pid_value = kwargs["pid_value"]
+            community = current_communities.service.read(
+                id_=pid_value, identity=g.identity
+            )
+            kwargs["community"] = community
```
[the remark this answers is not in the document]
Comment (jrcastro2) [reply invenio_communities/views/decorators.py:31]: Yes, we need to pass the `community` too because we also need to check for the permissions, see [here an example](https://github.com/inveniosoftware/invenio-communities/blob/5f46f455ccac721489b33e88b2877d07ae85feaa/invenio_communities/views/communities.py#L98).

File: invenio_communities/assets/semantic-ui/js/invenio_communities/settings/profile/index.js [hunk 6]
```diff
@@ -406,14 +405,15 @@ class CommunityProfileForm extends Component {
...
-
     // Serialize each funding record, award being optional.
-    const funding = submittedCommunity.metadata?.funding.map((fund) => {
-      return {
-        ...(fund.award && { award: serializeFunding(fund.award) }),
-        funder: serializeFunding(fund.funder),
-      };
-    });
+    const funding =
+      submittedCommunity.metadata?.funding &&
```
Comment (zzacharo) [inline invenio_communities/assets/semantic-ui/js/invenio_communities/settings/profile/index.js:410]: this is redundant as we use the `?.`...is there an issue?

Comment (jrcastro2) [reply invenio_communities/assets/semantic-ui/js/invenio_communities/settings/profile/index.js]: thanks!

**Review by zzacharo — APPROVED** [review]
